acceptanceTests/cypress/integration/tests/UiTest.js

```diff
@@ -182,39 +182,53 @@ function checkIllegalFilter(illegalFilterName) {
 
 function checkFilter(filterDetails) {
     const {
-        name,
+        filter,
         leftSidePath,
         rightSidePath,
         rightSideExpectedText,
         leftSideExpectedText,
-        applyByEnter
+        applyByCtrlEnter
     } = filterDetails;
 
     const entriesForDeeperCheck = 5;
 
-    it(`checking the filter: ${name}`, function () {
+    it(`checking the filter: ${filter}`, function () {
+        // wait half a second and pause the stream to preserve the DOM
+        cy.wait(500);
+        cy.get('#pause-icon').click();
+        cy.waitUntil(function() {
```

Comment:
why is this needed? 
cy.get('#pause-icon').should('not.be.visible') waits for the pause icon to not be visible

Reply:
@RoyUP9 I didn't want to add more recursive promises and further complicate the code. So, I used https://www.npmjs.com/package/cypress-wait-until instead;

> **Why did we write it?**
A lot of StackOverflow users had some difficulties in implementing a recursive promise with Cypress just to repeatedly check for something to happen (see two of the various questions about the topic: How can i wait for each element in a list to update to a certain text? And How do I wait until a cookie is set?).
This plugin is dedicated to them ❤️

Reply:
Isn't cypress default behavior to wait X timeout for the should part to be true (without the waitUntill)?

Reply:
@RoyUP9 `cypress-wait-until` dependency is removed with https://github.com/up9inc/mizu/commit/4f0a67feca968103f1a00afd2bed5a4508129862